src/adapters/webpush/db.rs

```diff
@@ -0,0 +1,215 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use super::Subscription;
+use libc::c_int;
+use rusqlite::{ self, Connection };
+
+fn escape(string: &str) -> String {
+    // http://www.sqlite.org/faq.html#q14
+    string.replace("'", "''")
+}
+
+pub struct WebPushDb {
+    db: Connection,
+}
+
+impl WebPushDb {
+    pub fn new(path: &str) -> Self {
+        let db = Connection::open(path).unwrap();
+        db.execute("CREATE TABLE IF NOT EXISTS subscriptions (
+                    user_id     INTEGER,
+                    push_uri    TEXT NOT NULL UNIQUE,
+                    public_key  TEXT NOT NULL
+            )", &[]).unwrap();
+
+        db.execute("CREATE TABLE IF NOT EXISTS resources (
+                    user_id     INTEGER,
+                    resource    TEXT NOT NULL
+            )", &[]).unwrap();
+
+        WebPushDb {
+            db: db
+        }
+    }
+
+    pub fn subscribe(&self, user_id: i32, sub: &Subscription) -> rusqlite::Result<c_int> {
```

Comment:
nit: We should have a custom type for user_id, but that can be done in a followup.